store/tikv/client.go

```diff
@@ -372,7 +374,26 @@ func (c *RPCClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.R
 	if config.GetGlobalConfig().TiKVClient.MaxBatchSize > 0 && enableBatch {
 		if batchReq := req.ToBatchCommandsRequest(); batchReq != nil {
 			defer trace.StartRegion(ctx, req.Type.String()).End()
-			return sendBatchRequest(ctx, addr, req.ForwardedHost, connArray.batchConn, batchReq, timeout)
+			resp, err := sendBatchRequest(ctx, addr, req.ForwardedHost, connArray.batchConn, batchReq, timeout)
+			if err != nil {
+				return nil, err
+			}
+			execDetail := &kvrpcpb.ExecDetailsV2{}
+			var affectRow int
+			switch r := resp.Resp.(type) {
+			case kvrpcpb.GetResponse:
+				execDetail = r.ExecDetailsV2
+				affectRow = len(r.Value)
+			case kvrpcpb.BatchGetResponse:
+				execDetail = r.ExecDetailsV2
+				affectRow = len(r.Pairs)
+			}
+			if execDetail != nil {
+				readByte := execDetail.GetScanDetailV2().GetReadBytes()
+				readTime := float64(execDetail.GetTimeDetail().GetKvReadWallTimeMs())
+				sli.ObserveReadSLI(uint64(affectRow), readByte, readTime)
+			}
+			return resp, nil
```

Comment:
Why put that logic in here, it is too strange.

And it will duplicate with `distsql/select_result.go#306`